File: chap2/Elog.cpp
// Ces programmes sont sous licence CeCILL-B V1.
#include <iostream>

using namespace std;

int main()
{
   int n;
   double x;
   cin >> x;
   n = 0;
   while (x > 1) 
   {
      x = x / 2;
      n = n + 1;
   }
   cout << n << endl;
}

File: chap2/Elog.ml
(* Ces programmes sont sous licence CeCILL-B V1. *)

let _ =
    let x = float_of_string (input_line stdin) in
    let x = ref x in (* version modifiable de x *)
    let n = ref 0 in
    while !x > 1. do
      x := !x /. 2.;
      n := !n + 1;
    done;
    print_int !n; print_newline ()


